appcell-resources/js/settings.js

```diff
@@ -567,36 +590,46 @@ st.editAccountNameBlurEvent = function() {
 st.addAccount = function() {
   var name = $("#addAccountName").val();
   if (st.validateName(name, "popupAddAccountNameErrorMsg", "-_!\$\*=^`\{\|\}~.@", "")) {
-    var pass = $("#pAddNewPassword").val();
-    if (st.passInputCheck(pass, "addChangeMessage")
-     && st.changePassCheck(pass, $("#pAddConfirm").val(), "addConfirmMessage")) {
-        var jsonData = {
-                        "Name" : name
-        };
-
-        // Assigning Roles
-        var chkObj = document.getElementById("addCheckAccountLinkRole");
-        if (chkObj.checked) {
-          if (st.checkAccLinkRole()) {
-            st.linkAccName = name;
-            st.restCreateAccountAPI(jsonData, pass);
-            return true;
+      if ($("input[name=accType]:checked").val() == "typePassword") {
+          var pass = $("#pAddNewPassword").val();
+          if (st.passInputCheck(pass, "addChangeMessage")
+              && st.changePassCheck(pass, $("#pAddConfirm").val(), "addConfirmMessage")) {
+              var jsonData = {
+                  "Name": name
+              };
+
+              // Assigning Roles
+              var chkObj = document.getElementById("addCheckAccountLinkRole");
+              if (chkObj.checked) {
+                  if (st.checkAccLinkRole()) {
+                      st.linkAccName = name;
+                      st.restCreateAccountAPI(jsonData, pass);
+                      return true;
+                  }
+              } else {
+                  st.restCreateAccountAPI(jsonData, pass);
+                  return true;
+              }
           }
-        } else {
-          st.restCreateAccountAPI(jsonData, pass);
-          return true;
-        }
-    }
+      } else {
+          st.linkAccName = name;
+          var jsonData = {
+              "Name": name,
+              "Type": "oidc:google"
+          };
+          st.restCreateAccountAPI(jsonData);
+      }
   }
 
   return false;
 };
 st.validateEditedInfo = function() {
   var name = $("#editAccountName").val();
+  var type = $("#editAccountName").data('type');
   if (st.validateName(name, "popupEditAccountNameErrorMsg", "-_!\$\*=^`\{\|\}~.@", "")) {
     var pass = $("#pEditNewPassword").val();
-    if (st.passInputCheck(pass, "editChangeMessage")
-     && st.changePassCheck(pass, $("#pEditConfirm").val(), "editConfirmMessage")) {
+    if (type !== "basic" || (st.passInputCheck(pass, "editChangeMessage")
```

Comment:
The logic becomes complicated. It is better to do the following.  

    if (!checkCondition1) {
        return false;
    }
    if (!checkCondition2) {
        return false;
    }
    // Do something
   return false;